src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -0,0 +1,3129 @@
+                        }
+
+                    case RegexCode.Onelazy | RegexCode.Back:
+                    case RegexCode.Notonelazy | RegexCode.Back:
+                    case RegexCode.Setlazy | RegexCode.Back:
+                    case RegexCode.Onelazy | RegexCode.Rtl | RegexCode.Back:
+                    case RegexCode.Notonelazy | RegexCode.Rtl | RegexCode.Back:
+                    case RegexCode.Setlazy | RegexCode.Rtl | RegexCode.Back:
+                    case RegexCode.Onelazy | RegexCode.Ci | RegexCode.Back:
+                    case RegexCode.Notonelazy | RegexCode.Ci | RegexCode.Back:
+                    case RegexCode.Setlazy | RegexCode.Ci | RegexCode.Back:
+                    case RegexCode.Onelazy | RegexCode.Ci | RegexCode.Rtl | RegexCode.Back:
+                    case RegexCode.Notonelazy | RegexCode.Ci | RegexCode.Rtl | RegexCode.Back:
+                    case RegexCode.Setlazy | RegexCode.Ci | RegexCode.Rtl | RegexCode.Back:
+                        {
+                            const string I = "tmp1";
+
+                            writer.WriteLine($"runtextpos = {PopTrack()};");
+                            writer.WriteLine($"{I} = {PopTrack()}; // i");
+
+                            string expr = !IsRightToLeft() ?
+                                Rightcharnext() :
+                                Leftcharnext();
+
+                            if (Code() == RegexCode.Setlazy)
+                            {
+                                EmitTimeoutCheck(writer, hasTimeout);
+                                expr = $"!{MatchCharacterClass(hasTextInfo, options, expr, rm.Code.Strings[Operand(0)], IsCaseInsensitive())}";
+                            }
+                            else
+                            {
+                                expr = ToLowerIfNeeded(hasTextInfo, options, expr, IsCaseInsensitive() && RegexCharClass.ParticipatesInCaseConversion(Operand(0)));
+                                expr = $"{expr} {(Code() == RegexCode.Onelazy ? "!=" : "==")} {Literal((char)Operand(0))}";
+                            }
+
+                            using (EmitBlock(writer, $"if ({expr})"))
+                            {
+                                writer.WriteLine($"goto {Backtrack};");
+                            }
+
+                            using (EmitBlock(writer, $"if ({I} > 0)"))
+                            {
+                                PushTrack($"{I} - 1");
+                                PushTrack("runtextpos");
+                                Trackagain();
+                            }
+
+                            Advance();
+                        }
+                        break;
+
+                    default:
+                        Debug.Fail($"Unimplemented state: {currentOpcode:X8}");
+                        break;
+                }
+            }
+
+
+
+            /// <summary>
+            /// Branch to the MSIL corresponding to the regex code at i
```

Comment:
MSIL?

Reply:
It's just a copy/paste of RegexCompiler.  I'll change it.